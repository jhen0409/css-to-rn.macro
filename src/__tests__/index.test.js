const pluginTester = require('babel-plugin-tester')
const plugin = require('babel-plugin-macros')

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    'Convert styles successful': `
import { StyleSheet } from 'react-native';
import css from '../macro';
const styles = StyleSheet.create(
  css\`
    .container {
      flex: 1;
      justify-content: center;
      align-items: center;
    }
    .text {
      font-size: 18;
      color: black;
    }
  \`
);
    `,
    'Fallback to `css-to-react-native-transfom` import if used expressions': `
import { StyleSheet } from 'react-native';
import css from '../macro';
const height = 100;
const styles = StyleSheet.create(
  css\`
    .container {
      flex: 1;
      height: \${height};
      justify-content: center;
      align-items: center;
    }
    .text {
      font-size: 18;
      color: black;
    }
  \`
);
    `,
    'parseMedia successful': `
import { StyleSheet } from 'react-native'
import css, { parseMedia } from 'css-to-rn.macro'

const styles = StyleSheet.create(
  parseMedia(
    css\`
      .container {
        flex: 1;
        justify-content: center;
        align-items: center;
      }
      @media not android {
        .container {
          background-color: #ccc;
        }
      }
    \`,
    {
      /* match object of react-native-css-media-query-processor */
    },
  ),
)
parseMedia(css\`@media not android { .container { flex: 1; } }\`)
parseMedia
    `,
  },
})
