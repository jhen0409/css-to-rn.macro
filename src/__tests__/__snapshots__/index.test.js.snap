// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros Convert styles successful: Convert styles successful 1`] = `
"
import { StyleSheet } from 'react-native';
import css from '../macro';
const styles = StyleSheet.create(
  css\`
    .container {
      flex: 1;
      justify-content: center;
      align-items: center;
    }
    .text {
      font-size: 18;
      color: black;
    }
  \`
);

      ↓ ↓ ↓ ↓ ↓ ↓

import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  'container': {
    'flexGrow': 1,
    'flexShrink': 1,
    'flexBasis': 0,
    'justifyContent': 'center',
    'alignItems': 'center'
  },
  'text': {
    'fontSize': 18,
    'color': 'black'
  }
});
"
`;

exports[`macros Fallback to \`css-to-react-native-transfom\` import if used expressions: Fallback to \`css-to-react-native-transfom\` import if used expressions 1`] = `
"
import { StyleSheet } from 'react-native';
import css from '../macro';
const height = 100;
const styles = StyleSheet.create(
  css\`
    .container {
      flex: 1;
      height: \${height};
      justify-content: center;
      align-items: center;
    }
    .text {
      font-size: 18;
      color: black;
    }
  \`
);

      ↓ ↓ ↓ ↓ ↓ ↓

import _css from 'css-to-react-native-transform';
import { StyleSheet } from 'react-native';

const height = 100;
const styles = StyleSheet.create(_css(\`
    .container {
      flex: 1;
      height: \${height};
      justify-content: center;
      align-items: center;
    }
    .text {
      font-size: 18;
      color: black;
    }
  \`, {
  'parseMediaQueries': true
}));
"
`;

exports[`macros parseMedia successful: parseMedia successful 1`] = `
"
import { StyleSheet } from 'react-native'
import css, { parseMedia } from 'css-to-rn.macro'

const styles = StyleSheet.create(
  parseMedia(
    css\`
      .container {
        flex: 1;
        justify-content: center;
        align-items: center;
      }
      @media not android {
        .container {
          background-color: #ccc;
        }
      }
    \`,
    {
      /* match object of react-native-css-media-query-processor */
    },
  ),
)
parseMedia(css\`@media not android { .container { flex: 1; } }\`)
parseMedia

      ↓ ↓ ↓ ↓ ↓ ↓

import { process as _parseMedia } from 'react-native-css-media-query-processor';
import { StyleSheet } from 'react-native';


const styles = StyleSheet.create(_parseMedia({
  'container': {
    'flexGrow': 1,
    'flexShrink': 1,
    'flexBasis': 0,
    'justifyContent': 'center',
    'alignItems': 'center'
  },
  '__mediaQueries': {
    '@media not android': [{
      'inverse': true,
      'type': 'android',
      'expressions': []
    }]
  },
  '@media not android': {
    'container': {
      'backgroundColor': '#ccc'
    }
  }
}, {
  /* match object of react-native-css-media-query-processor */
}));
_parseMedia({
  '__mediaQueries': {
    '@media not android': [{
      'inverse': true,
      'type': 'android',
      'expressions': []
    }]
  },
  '@media not android': {
    'container': {
      'flexGrow': 1,
      'flexShrink': 1,
      'flexBasis': 0
    }
  }
}, {});
parseMedia;
"
`;
